app/vmauth/main.go

```diff
@@ -147,23 +147,38 @@ func processUserRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 
 func processRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 	u := normalizeURL(r.URL)
-	up, headers, err := ui.getURLPrefixAndHeaders(u)
-	if err != nil {
-		httpserver.Errorf(w, r, "cannot determine targetURL: %s", err)
-		return
+	up, headers := ui.getURLPrefixAndHeaders(u)
+	isDefault := false
+	if up == nil {
+		missingRouteRequests.Inc()
+		if ui.DefaultURL == nil {
+			httpserver.Errorf(w, r, "missing route for %q", u.String())
+			return
+		}
+		up, headers = ui.DefaultURL, ui.Headers
+		isDefault = true
 	}
+
 	maxAttempts := up.getBackendsCount()
 	for i := 0; i < maxAttempts; i++ {
 		bu := up.getLeastLoadedBackendURL()
-		targetURL := mergeURLs(bu.url, u)
+		targetURL := bu.url
+		// Don't change path and add request_path query param for default route.
```

Comment:
Why targetURL shouldn't  be merged? I think it should be merged.

 E.g. for request /api/v1/query with  default_url: http://localhost:8081/proxy should be merged into `http://localhost:8081/proxy/api/v1/query`